### PR TITLE
Implement the AO API in torchchat quantization handlers and unify logic.

### PR DESCRIPTION
**Pull request by mikekgfb**
Implement the AO API in torchchat quantization handlers and unify logic.

1 - implement .quantize() for TC quantization handlers and support args to make consistent with AO
2 - remove special handling for various combinations of parameters and use validate_args before calling with **q_kwargs
3 - remove check probing whether we successfully loaded a8wx and install an error-reporting handler if loading failed which will be called as quant handler and issue an error
4 - unify both tc and ao quantization handler dicts with shared calling logic
5 - provide informational message when quantizer option not supported (via introspection)

### CODE REVIEW

File: torchchat/utils/quantize.py [hunk 1]
```diff
@@ -890,7 +901,7 @@ def quantized_model(self) -> nn.Module:
     sys.modules["torchao_experimental_quant_api"] = torchao_experimental_quant_api
     torchao_experimental_quant_api_spec.loader.exec_module(torchao_experimental_quant_api)
     from torchao_experimental_quant_api import Int8DynActIntxWeightQuantizer
-    ao_quantizer_class_dict["linear:a8wxdq"] = Int8DynActIntxWeightQuantizer
+    quantizer_class_dict["linear:a8wxdq"] = Int8DynActIntxWeightQuantizer
```
Comment (larryliu0820) [inline torchchat/utils/quantize.py:904]: Any reason we can't put this inside of the `quantizer_class_dict`?

Comment (larryliu0820) [reply torchchat/utils/quantize.py:904]: It seems cleaner to try import `Int8DynActIntxWeightQuantizer` first and fallback to `ErrorHandler`, then assign this handler to `linear:a8wxdq` in `quantizer_class_dict`.

**Review by larryliu0820 — CHANGES_REQUESTED** [review]
Thanks for the cleanup! Just one comment and please make sure CI jobs are passing

**Comment by mikekgfb** [comment]
> Thanks for the cleanup! Just one comment and please make sure CI jobs are passing

**Comment by mikekgfb** [comment]
> > Thanks for the cleanup! Just one comment and please make sure CI jobs are passing

My pleasure.  Sorry about that error, should be fixed now.

File: torchchat/utils/quantize.py (hunk 1 above)
Comment (larryliu0820) [reply torchchat/utils/quantize.py:903]: I think if this can go away soon I don't have a strong opinion and can live with this.

**Review by larryliu0820 — APPROVED** [review]
CI is passing

**Pull request merged**